ports/proj4/portfile.cmake

```
@@ -63,5 +64,44 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
 vcpkg_copy_pdbs()
 
+set(PACKAGE "${PORT}")
+set(requires_private "sqlite3")
+if(ENABLE_CURL)
+    string(APPEND requires_private " libcurl")
+endif()
+if(ENABLE_TIFF)
+    string(APPEND requires_private " libtiff-4")
+endif()
+set(libs_private "")
+if("c\\+\\+" IN_LIST VCPKG_SYSTEM_LIBRARIES)
```

```suggestion
if("c++" IN_LIST VCPKG_SYSTEM_LIBRARIES)
```

Since `IN_LIST` doesn't do a regex match, this is the correct way to check. (same for `stdc++` below)

---

@strega-nil-ms Once again, it seems like part of what I wrote went unnoticed. Let me repeat:

> However, this variable doesn't have the literal name (`stdc++` etc.) but escapes + (`stdc\+\+`). Is this intentional?

And the source is:
https://github.com/microsoft/vcpkg/blob/b6b6a8b63007df2ff37167a1974652a8e948f045/scripts/cmake/vcpkg_common_definitions.cmake#L173-L181

---

Okay, it seems like `VCPKG_SYSTEM_LIBRARIES` isn't used anymore anywhere; marking requires:discussion so we can discuss how we want to do this.

---

In case you are going to make changes:
It is not enough that triplets may add libraries. They need to override. E.g. for android, both c++ and stdc++ are possible alternatives.